tests/comparisons/Migration/testCompositeConstraintsSnapshot.php

```diff
@@ -34,26 +34,31 @@ public function up()
             ])
             ->addIndex(
                 [
-                    'slug',
-                ],
-                ['unique' => true]
+                    'category_id',
+                ]
+            )
+            ->addIndex(
+                [
+                    'product_id',
+                ]
```

Comment:
@markstory I double checked this and it turns out it is the only way to have something working whatever the case.
The problem I have still occurs and is due to the fact that the ``orders`` table is referencing the ``product_category`` column from ``products`` that is not referenced as an index if the ``addIndex()`` calls are not done before trying to create the constraints. MySQL seems not to be able to create a constraint with columns that are not referenced as index and raises an exception if I try to migrate without the ``addIndex()`` calls.

There is no problem for the other tables because all other constraints are done on the id column which is a PK and already referenced as an index as such.